src/telemetry.ts

```diff
@@ -100,38 +49,31 @@ export class Telemetry {
 	}
 
 	/**
-	 * Send a telemetry event for an exception
+	 * Send an Error telemetry event
 	 */
-	public static sendTelemetryEventForException(
-		err: any, methodName: string, extensionConfigName: string): void {
-		try {
-			let stackArray: string[];
-			let firstLine: string = '';
-			if (err !== undefined && err.stack !== undefined) {
-				stackArray = err.stack.split('\n');
-				if (stackArray !== undefined && stackArray.length >= 2) {
-					firstLine = stackArray[1]; // The fist line is the error message and we don't want to send that telemetry event
-					firstLine = FilterErrorPath(firstLine);
-				}
-			}
 
-			// Only adding the method name and the fist line of the stack trace. We don't add the error message because it might have PII
-			this.sendTelemetryEvent('Exception', { methodName: methodName, errorLine: firstLine });
-			// Utils.logDebug('Unhandled Exception occurred. error: ' + err + ' method: ' + methodName, extensionConfigName);
+	public static sendErrorTelemetry( view: string, name: string, errorcode: string, errortype: string, providername: string ): void {
+		if (typeof this.disabled === 'undefined') {
+			this.disabled = false;
+		}
+
+		if (this.disabled || typeof (this.reporter) === 'undefined') {
+			// Don't do anything if telemetry is disabled
+			return;
+		}
```

Comment:
You don't need to do this yourself - the underlying VS Code extension library handles checking the enabled state : https://github.com/microsoft/vscode-extension-telemetry/blob/a889f4c73d53c2680c9073ab5b7b46e57a2fbe5a/src/common/util.ts#L40

Reply:
Here's an example PR of me cleaning this up in our own extensions : https://github.com/microsoft/azuredatastudio/pull/21779/files?show-viewed-files=true&file-filters%5B%5D=#diff-5a9148b7d1354d9f787120de87ff5d4c63c79429a0d0d184eba2638f3e4865dd

Reply:
removed them.